ping/ping6_common.c

```diff
@@ -304,7 +303,7 @@ int ping6_run(struct ping_rts *rts, int argc, char **argv, struct addrinfo *ai,
 		rts->timing = 1;
 	}
 	packlen = rts->datalen + 8 + 4096 + 40 + 8; /* 4096 for rthdr */
-	if (!(packet = (unsigned char *)malloc((unsigned int)packlen)))
+	if (!(packet = malloc((unsigned int)packlen)))
```

Comment:
The cast to unsigned int here is useless as well. We are passing size_t here now which is the proper type to be passed to malloc().